certbot/main.py

```diff
@@ -859,6 +861,45 @@ def plugins_cmd(config, plugins):
     logger.debug("Prepared plugins: %s", available)
     notify(str(available))
 
+def enhance(config, plugins):
+    """Add security enhancements to existing configuration
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param plugins: List of plugins
+    :type plugins: `list` of `str`
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    supported_enhancements = ["hsts", "redirect", "uir", "staple"]
+    # Check that at least one enhancement was requested on command line
+    if not any([getattr(config, enh) for enh in supported_enhancements]):
+        msg = ("Please specify one or more enhancement types to configure. To list "
+               "the available enhancement types, run:\n\n%s --help enhance\n")
+        logger.warning(msg, sys.argv[0])
+        raise errors.MisconfigurationError("No enhancements requested, exiting.")
+
+    try:
+        installer, _ = plug_sel.choose_configurator_plugins(config, plugins, "enhance")
+    except errors.PluginSelectionError as e:
+        return str(e)
+
+    if not config.certname:
+        config.certname = cert_manager.get_certnames(config, "enhance",
```

Comment:
This causes the prompt to be "Which certificate would you like to enhance?". Perhaps it should be something like "Which certificate would you like to use to enhance your configuration?".

This may mean we may want to change the name of the `verb` parameter to `get_certnames`.

Reply:
`get_certnames` now takes an optional kwarg if the prompt message needs to be changed completely.